### PR TITLE
[ADD] inventory: add before date res doc

### PR DESCRIPTION
**Pull request by brse-odoo**
This PR is an [ADD] to add a `before_scheduled_date` doc into Odoo's Inventory documentation, nested under the `shipping_receiving` section, and the `reservation_methods` section, specific to 17.0.

Here's what I pushed up in this one:

- New doc for third of 4 reservation methods docs
- Updated toc-tree and added new image folder for doc

**_Important:_**

- _Please ignore any conflicts, as those are expected and will be resolved as I continue to push up changes to all 4 PRs._

### Merge order:
Merge this PR after [PR #7886](https://github.com/odoo/documentation/pull/7886).

### CODE REVIEW

**Comment by brse-odoo** [comment]
@odoo/inventory-doc-review This PR is ready for review. Per commit message, please ignore any merge conflicts for the time being. Thank you in advance!

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:33]:
```suggestion
Changing the :guilabel:`days before when starred` value changes the maximum number of days before
a scheduled date that products should be reserved for starred (favorited) transfers.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:74]:
```suggestion
Click the green :guilabel:` 📈 (forecast graph)` icon on the product line to reveal the product's
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:102]:
```suggestion
:guilabel:`Demand` column and the :guilabel:`Quantity` column do not match (in this case, the
```

**Review by hojo-odoo — APPROVED** [review]
Hey @brse-odoo,

Nice job on this. I just had a few comments for you to look at before you move it to the next round of review.

Thanks,
hojo

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:33]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:74]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:102]: Suggestion implemented ✅

**Comment by brse-odoo** [comment]
Thank you @hojo-odoo for the quick review!

Alright @odoo/us-doc-review, this PR has passed the peer review round, and is now ready for the final content review round. Please feel free to reach out to me if you have any questions. Thank you!

**_Note:_** The doc that needs to be reviewed in this PR is the `before_scheduled_date.rst` doc. Thank you!

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (ksc-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:91]: removed the GUI from 'Delivery Order' because that isn't shown/displayed on the screen/UI

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:112]:
```suggestion
   changes the number in the :guilabel:`Quantity` column to match the :guilabel:`Demand` column.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (ksc-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:103]: Considering this is where the sample flow ends currently, it seems a bit "unfinished." Perhaps you could add a mention of how, when the time comes, the numbers in the columns will match...? Or something along those lines. It just seems like a bit of "conclusion" is needed to round things out a touch.

**Review by ksc-odoo — CHANGES_REQUESTED** [review]
@brse-odoo hey there! Just finished my Final Review of this doc. Great work! There is a handful of comments that require your attention, but once you make the necessary adjustments, tag me again for another quick look. Thanks! 👍

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/manually.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/manually.rst:109]: Yep, all set (doc will update when I rebase after making edits to this PR). Thank you 🙏

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:7]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:19]: Yes, I added in a note admonition explaining this:

.. note::
   When the :guilabel:`Type of Operation` is changed to *Receipt* on an :guilabel:`Operations Type`
   form, reservation methods are **not** available.
   
  The only other "restrictions" would be settings that need to be enabled for other operations types to appear, which is outlined in the Reservation Methods overview/parent doc (linked above this).

Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:40]: I didn't know that before, but now I do 👍 Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:52]: Yeah, that's what I meant to convey here, so the clarifying language is helpful. Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:56]: I understand that logic, it does make sense in this instance. Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:70]: Sounds good to me. Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:81]: Yes, that is definitely helpful to clarify - I added in the following admonition to explain:

.. note::
   If there is **not** sufficient quantity of stock for the product included in the |SO|, the
   :guilabel:`📈 (forecast graph)` icon is red, instead of green.

   Instead of revealing the reserved number of units for the order, the :guilabel:`Availability`
   tooltip reads :guilabel:`Reserved`, and reveals the available number of units (e.g., `0 Units`).

   Additionally, unless there is a set replenishment or a live receipt, it also reads :guilabel:`No
   future availability`, in red text.

Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:91]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:95]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:108]: Yeah, I just thought it would be useful to indicate that it can apply to both orders with 1 product, and orders with multiple. But I'll stick to the rules here, Suggestion Implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:112]: Suggestion implemented ✅

**Comment by brse-odoo** [comment]
> @brse-odoo hey there! Just finished my Final Review of this doc. Great work! There is a handful of comments that require your attention, but once you make the necessary adjustments, tag me again for another quick look. Thanks! 👍

Hi @ksc-odoo, thanks for the quick review. Per your request I'm re-tagging you for a second review when you have a chance. Thanks!

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:32]:
```suggestion
   When the :guilabel:`Type of Operation` is changed to :guilabel:`Receipt` on an
   :guilabel:`Operations Type` form, reservation methods are **not** available.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:42]:
```suggestion
a scheduled date that products should be reserved if the transfers are starred (favorited).
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:46]:
```suggestion
   If the :guilabel:`days before` value is set to `2` days before, and the :guilabel:`days before
   when starred` value is set to `3`.
```
I know you implemented my previous suggestion, but I think this current one is a bit better -- your call.

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:55]:
```suggestion
   This is the configuration applied for the following workflow found below.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:86]:
```suggestion
:guilabel:`Add a product`, and select a product from the drop-down menu that has a configured
*customer lead time*, to add to the quotation form.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:103]:
```suggestion
Since reservation is not until two days before the scheduled delivery, it will not reserve the
products until then.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:135]:
```suggestion
:guilabel:`Demand` column lists `10.00`, while the :guilabel:`Quantity` column lists `0`).
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:142]:
```suggestion
   If the products in the |SO| should be reserved *sooner* than the scheduled reservation date, the
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:143]:
```suggestion
   reservation can be manually overridden. To manually reserve the products sooner than scheduled,
```

**Review by ksc-odoo — APPROVED** [review]
Great work on implementing the changes, @brse-odoo -- i only have a couple more comments that require your attention, but I'm approving now, because I don't think you'll have any issues with them. Once you've made the necessary adjustments, feel free to tag this for Tech Review. Thanks!

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:32]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:42]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:46]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:86]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:103]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:135]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:143]: Suggestion implemented ✅

**Comment by brse-odoo** [comment]
Thank you @ksc-odoo for the review!

**Comment by brse-odoo** [comment]
**_PR Notes:_** PR has passed peer review and us-doc-review stages. Next up is technical review - however, the PR before this one will need to be merged before I can retarget 17.0 as the base branch and fix the failing CI check. Will do so once I can, and tag technical review.

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:42]: Early break :)
```suggestion
Changing the :guilabel:`days before when starred` value changes the maximum number of days before a
scheduled date that products should be reserved if the transfers are starred (favorited).
```

**Review by samueljlieber — CHANGES_REQUESTED** [review]
Hi @brse-odoo this PR looks great to me, while we are waiting on the first two PRs to fully merge, here is one small suggestion. Please tag me for another quick look once this PR has been rebased after both dependent PRs are fully merged. thank you!

**Comment by StraubCreative** [comment]
@samueljlieber @brse-odoo #8190 has been resolved to complete the chain for #7886 

For this PR (After #8190 finishes it's deployment), we'll need to: 
1. reset it to the last commit that *only* contained the additions surrounding `before_scheduled_date.rst`
2. rebase onto fresh 17.0 pull to incorporate all of the previous merges and remove merge the conflicts on tertiary files
3. reapply, as necessary, all of the targeted edits from the review rounds in this PR
4. squash all commits, rebase again on fresh 17.0 pull, and force push (overwrite)

From there we (Sam and I) can spot check that all CRs were addressed, and there are no more merge conflicts 👍 
Almost there for this series, let's finish strong! 💪

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:42]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:153]: No need to back out of this directory and move right back into it 😄, if I suggested this in the past I apologize, the path can simply be:
```suggestion
   - :doc:`Manual reservation <manually>`
   - :doc:`At confirmation reservation <at_confirmation>`
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/manually.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/manually.rst:139]:
```suggestion
   - :doc:`At confirmation reservation <at_confirmation>`
   - :doc:`Before scheduled date reservation <before_scheduled_date>`
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:90]:
```suggestion
   - :doc:`Manual reservation <manually>`
   - :doc:`Before scheduled date reservation <before_scheduled_date>`
```

**Review by samueljlieber — CHANGES_REQUESTED** [review]
Hi @brse-odoo, awesome job rebasing this PR after all the previous deployments, it looks like all previous corrections persisted 👍. I have a small fix to update the doc paths, but everything else in this PR looks good to me! Please tag me for one more quick look after this is addressed, thank you!

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/before_scheduled_date.rst:153]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/manually.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/manually.rst:139]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst:90]: Suggestion implemented ✅

**Review by samueljlieber — APPROVED** [review]
Thank you @brse-odoo, looks good to me 👍

**Comment by brse-odoo** [comment]
> Thank you @brse-odoo, looks good to me 👍

Awesome, thank you @samueljlieber! @StraubCreative This PR is ready for your final lookover and approval.

**Review by StraubCreative — APPROVED** [review]
Approved, thanks @brse-odoo 👍 
.....
@robodoo r+

**Pull request closed without merging**
